Latest dill release raises exception

## Describe the bug
As reported by @sgugger, latest dill release is breaking things with Datasets.

```
______________ ExamplesTests.test_run_speech_recognition_seq2seq _______________


self = <multiprocess.pool.ApplyResult object at 0x7fa5981a1cd0>, timeout = None

    def get(self, timeout=None):
        self.wait(timeout)
        if not self.ready():
            raise TimeoutError
        if self._success:
            return self._value
        else:
>           raise self._value
E           TypeError: '>' not supported between instances of 'NoneType' and 'float'
```

--- a/setup.py
+++ b/setup.py
@@ -67,7 +67,7 @@ REQUIRED_PKGS = [
     # Minimum 6.0.0 to support wrap_array which is needed for ArrayND features
     "pyarrow>=6.0.0",
     # For smart caching dataset processing
-    "dill",
+    "dill<=0.3.1.1",  # tmp pin: see https://github.com/huggingface/datasets/issues/4379
     # For performance gains with apache arrow
     "pandas",
     # for downloading datasets over HTTPS
@@ -79,7 +79,7 @@ REQUIRED_PKGS = [
     # for fast hashing
     "xxhash",
     # for better multiprocessing
-    "multiprocess",
+    "multiprocess<=0.70.9",  # tmp pin: see https://github.com/huggingface/datasets/issues/4379
     # to get metadata of optional dependencies such as torch or tensorflow for Python versions that don't have it
     "importlib_metadata;python_version<'3.8'",
     # to save datasets locally or on any filesystem

--- a/setup.py
+++ b/setup.py
@@ -79,7 +79,7 @@ REQUIRED_PKGS = [
     # for fast hashing
     "xxhash",
     # for better multiprocessing
-    "multiprocess<=0.70.9",  # tmp pin: see https://github.com/huggingface/datasets/issues/4379
+    "multiprocess",
     # to get metadata of optional dependencies such as torch or tensorflow for Python versions that don't have it
     "importlib_metadata;python_version<'3.8'",
     # to save datasets locally or on any filesystem

--- a/setup.py
+++ b/setup.py
@@ -67,7 +67,7 @@ REQUIRED_PKGS = [
     # Minimum 6.0.0 to support wrap_array which is needed for ArrayND features
     "pyarrow>=6.0.0",
     # For smart caching dataset processing
-    "dill<=0.3.1.1",  # tmp pin: see https://github.com/huggingface/datasets/issues/4379
+    "dill<=0.3.4",  # tmp pin: see https://github.com/huggingface/datasets/issues/4379
     # For performance gains with apache arrow
     "pandas",
     # for downloading datasets over HTTPS

--- a/setup.py
+++ b/setup.py
@@ -67,7 +67,7 @@ REQUIRED_PKGS = [
     # Minimum 6.0.0 to support wrap_array which is needed for ArrayND features
     "pyarrow>=6.0.0",
     # For smart caching dataset processing
-    "dill<=0.3.4",  # tmp pin: see https://github.com/huggingface/datasets/issues/4379
+    "dill<0.3.5",  # tmp pin: see https://github.com/huggingface/datasets/issues/4379
     # For performance gains with apache arrow
     "pandas",
     # for downloading datasets over HTTPS

